staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -228,11 +228,13 @@ func PatchResource(r rest.Patcher, scope *RequestScope, admit admission.Interfac
 		}
 		trace.Step("Object stored in database")
 
-		if err := setObjectSelfLink(ctx, result, req, scope.Namer); err != nil {
-			scope.err(err, w, req)
-			return
+		if !utilfeature.DefaultFeatureGate.Enabled(features.RemoveSelfLink) {
+			if err := setObjectSelfLink(ctx, result, req, scope.Namer); err != nil {
+				scope.err(err, w, req)
+				return
+			}
+			trace.Step("Self-link added")
```

Comment:
Hmm - actually looking into it - why do we need to explicitly calls setObjectSelflink in the "patch" handler and not in any other operation (create, update, get)?

Is it an actual duplication? Can we get rid of it completely?